xcodeproj/internal/files/incremental_input_files.bzl

```diff
@@ -251,8 +251,17 @@ def _process_files_and_deps(
             ),
         )
 
+        # We need to collect `headers` in `generated` in addition to the above
+        # collection of `direct_{public,private}_headers` to get all files
+        # needed for indexing, even if they are from unsupported transitive
+        # dependencies
+        transitive_generated = [compilation_context.headers]
```

Comment:
If we leave it like this, we will have lower memory usage (since it will just reuse a depset that already exists), but we will have a larger BEP output. Checking how much larger now.

Reply:
I'm actually seeing lower BEP size using the depset. Probably because the repeated files are reused in `namedSet`s instead of being re-listed in each target's new `generated` `depset`.

Reply:
Some external testing suggests this is still a large change in the bep.

I think I'll make another change first that only includes these files for Index Build during BwB, since that's the only place they are needed, and Index Build typically won't have BEP collection enabled.